js/extension/components/FeatureReports.jsx

```diff
@@ -68,8 +75,23 @@ class FeatureReports extends React.Component {
 
     componentDidMount() {
         this.subscription = reportService
-            .getReports(this.props.feature.id)
-            .subscribe((reports) => this.setState({ reports }));
+            .getReports(this.state.feature_id, this.state.layer_id)
+            .subscribe((reports) => 
+                {
+                    let reports_parsed = reports;
+                    if (! new URLSearchParams(window.location.search).has("usemocks")) { 
+                        reports_parsed = reports.map((report) => JSON.parse(report));
+                    } 
+                    this.setState({ reports : reports_parsed })
+                });
+    }
+
+    postReport(payload) {
+        this.props.postReport(payload);
+        this.setState({
+            editReport: false,
+        });
+        this.componentDidMount();
```

Comment:
you should call `componentDidMount` manually, it's react internals. You'd better externalize its content.
```js
getReports() {
  this.sub = reportService…
}

componentDidMount() {
  this.getReports()
}
postReport() {
  …
  this.getReports()
}
```